### PR TITLE
VideoBackends:D3D12: Fix hang in Twilight Princess

### PR DESCRIPTION
**Pull request by K0bin**
Fixes [issue 13087](https://bugs.dolphin-emu.org/issues/13087)

The last commit is the interesting one. I explain the bug in the commit message.

### CODE REVIEW

**Comment by K0bin** [comment]
I adjusted the fifo diff for visibility:
image

I didn't change anything in the OpenGL backend, so that's gotta be unrelated to the PR.

**Comment by JMC47** [comment]
Sometimes there is noise in the fifoci bots, wouldn't worry about it.  Ghost in the machine type thing.

File: Source/Core/VideoBackends/D3D12/D3D12BoundingBox.cpp [hunk 1]
```diff
@@ -101,8 +101,8 @@ bool D3D12BoundingBox::CreateBuffers()
                                      D3D12_RESOURCE_FLAG_ALLOW_UNORDERED_ACCESS};
 
   HRESULT hr = g_dx_context->GetDevice()->CreateCommittedResource(
-      &gpu_heap_properties, D3D12_HEAP_FLAG_NONE, &buffer_desc,
-      D3D12_RESOURCE_STATE_UNORDERED_ACCESS, nullptr, IID_PPV_ARGS(&m_gpu_buffer));
+      &gpu_heap_properties, D3D12_HEAP_FLAG_NONE, &buffer_desc, D3D12_RESOURCE_STATE_COMMON,
```
Comment (iwubcode) [inline Source/Core/VideoBackends/D3D12/D3D12BoundingBox.cpp:104]: Do you recall what the debug layer error was?  It'd be nice if that could go into the commit

Comment (iwubcode) [reply Source/Core/VideoBackends/D3D12/D3D12BoundingBox.cpp:104]: Would be nice to do for the other one as well.

Comment (K0bin) [reply Source/Core/VideoBackends/D3D12/D3D12BoundingBox.cpp:104]: Resources in the READBACK heap must use this specific initial state.

Comment (iwubcode) [reply Source/Core/VideoBackends/D3D12/D3D12BoundingBox.cpp:104]: I meant add the debug layer error to the commit message (for the "don't query the GPU descriptor" commit)

**Review by iwubcode — APPROVED** [review]
Code LGTM.  I did not see any issues when I tested a handful of games (including the game impacted in the issue report).

**Pull request merged**